docs/rules/require-default-props.md

```diff
@@ -197,13 +197,23 @@ NotAComponent.propTypes = {
 
 ```js
 ...
-"react/require-default-props": [<enabled>, { forbidDefaultForRequired: <boolean>, ignoreFunctionalComponents: <boolean> }]
+"react/require-default-props": [<enabled>, {
+  "forbidDefaultForRequired": <boolean>,
+  // For now, this wouldn't support any other choices, and would be optional,
+  // but later could be set to "ignore" as long as `functions` wasn't set to "ignore"
+  "classes": "defaultProps",
+  "functions": "defaultProps" | "defaultArguments" | "ignore"
+  // @deprecated Use `functions` option instead.
+  "ignoreFunctionalComponents": <boolean>,
+}]
 ...
 ```
 
-* `enabled`: for enabling the rule. 0=off, 1=warn, 2=error. Defaults to 0.
-* `forbidDefaultForRequired`: optional boolean to forbid prop default for a required prop. Defaults to false.
-* `ignoreFunctionalComponents`: optional boolean to ignore this rule for functional components. Defaults to false.
+- `enabled`: for enabling the rule. 0=off, 1=warn, 2=error. Defaults to 0.
+- `forbidDefaultForRequired`: optional boolean to forbid prop default for a required prop. Defaults to false.
+- `classes`: For now, only works with "defaultProps".
+- `functions`: optional string that this rule determines strategy for the function. "defaultArguments" is recommended. We will change it to deafult later. Defaults to "defaultProps".
```

Comment:
defaultArguments is very much NOT recommended; defaultProps is a far superior mechanism.
```suggestion
- `functions`: optional string to determine which strategy a functional component uses for defaulting props. Defaults to "defaultProps".
```

Reply:
According to the original issue:
> defaultProps on function components is getting deprecated by React (see: facebook/react#16210)

It has been three-ish years since the original issue was reported. Is there any reason why the `defaultProps` is better now?

Reply:
Three-ish years and no actual deprecation, yes.

The primary reason is the same: `.defaultProps` are introspectable from outside the component, and default arguments are not.

Reply:
Oh, I didn't think that far. Thank you.